hack/verify-license.py

```diff
@@ -0,0 +1,945 @@
+#!/usr/bin/env python
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import fnmatch
+import difflib
+import hashlib
+import os
+import re
+import sys
+
+# How far a match can be from an expected license.
+MAX_EDIT_DISTANCE = 300
+
+# We have multiple variants of Apache License formatting.
+# These component parts allow them to be combined.
+LICENSE_APACHE_BASE = '''
```

Comment:
Can we move each license to a separate file, and use a `--flag=/path/to/dir-with-allowed-licesnes`?

Reply:
That would be a good change if this became standalone.